cmd/cluster/health.go

```diff
@@ -194,8 +268,56 @@ func createHealthObject(cluster *v1.Cluster) *ClusterHealthCondensedObject {
 	healthObject.Name = cluster.Name()
 	healthObject.Provider = cloudProviderMessage
 	healthObject.AZs = cluster.Nodes().AvailabilityZones()
+	healthObject.Hypershift = cluster.Hypershift().Enabled()
 	healthObject.Expected.Infra = cluster.Nodes().Infra()
 	healthObject.Expected.Master = cluster.Nodes().Master()
 
 	return &healthObject
 }
+
+func createHypershiftHealthObject(cluster *v1.Cluster) *ClusterHealthHypershiftObject {
+
+	var hsHealthObject ClusterHealthHypershiftObject
+
+	cloudProvider := cluster.CloudProvider().ID()
+	cloudProviderMessage := strings.ToUpper(cloudProvider)
+
+	hsHealthObject.Hypershift = cluster.Hypershift().Enabled()
+	hsHealthObject.ID = cluster.ID()
+	hsHealthObject.Name = cluster.Name()
+	hsHealthObject.Provider = cloudProviderMessage
+	hsHealthObject.AZs = cluster.Nodes().AvailabilityZones()
+
+	return &hsHealthObject
+}
+func getNodepools(clusterID string) ([]byte, error) {
+	get_nodepool := fmt.Sprintf("oc get nodepool -n ocm-production-%v", clusterID)
+
+	nodepool, err := exec.Command("bash", "-c", get_nodepool).CombinedOutput()
+	if err != nil {
+		return nil, fmt.Errorf("Failed to execute:\n%s", strings.TrimSpace(string(nodepool)))
+	}
+	return nodepool, nil
+}
+func getPods(environment string, clusterID string, clusterName string) ([]byte, error) {
+
+	get_po := fmt.Sprintf("oc get po -n ocm-%v-%v-%v| grep -v \"Running\\|Completed\" ", environment, clusterID, clusterName)
+
+	pods, err := exec.Command("bash", "-c", get_po).CombinedOutput()
```

Comment:
If we're already using golang, would you be opposed to getting the pods with native go libs from the oc sdk?  The more self-contained command is going to be more predictable than calling out to a bash command with multiple pipes and tons of potential errors generated from outside the go libs.